src/frontends/pytorch/src/op/one_hot.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/one_hot.hpp"
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/reduce_max.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/add.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_one_hot(const NodeContext& context) {
+    num_inputs_check(context, 1, 2);
+    auto x = context.get_input(0);
+    // aten::one_hot works on LongTensor which means we need to convert all inputs to i64
+    x = context.mark_node(std::make_shared<v0::Convert>(x, element::i64));
+    auto on_value = context.mark_node(v0::Constant::create(element::i64, Shape{}, {1}));
+    auto off_value = context.mark_node(v0::Constant::create(element::i64, Shape{}, {0}));
+    Output<Node> num_classes;
+    if (context.input_is_none(1)) {
+        num_classes = context.mark_node(v0::Constant::create(element::i64, Shape{}, {-1}));
+    } else {
+        num_classes = context.get_input(1);
+    num_classes = context.mark_node(std::make_shared<v0::Convert>(num_classes, element::i64));
+    }
+    auto zero = context.mark_node(v0::Constant::create(element::i64, Shape{}, {0}));
```

Comment:
It might be less descriptive, however, we could reuse on- and off-values as zero and one (ie, auto zero = off-value) to avoid creating multiple constants

Reply:
Agree, good to reuse